tox.ini

```diff
@@ -11,6 +11,24 @@ commands = pytest \
            --cov={envsitepackagesdir}/girder \
            {posargs}
 
+[testenv:docs]
+deps =
+    Sphinx
+    sphinx_rtd_theme
+# readthedocs.org runs "python setup.py install" (which installs no dependencies); with the
+# limitations of Tox (not being able to use a different install_command for deps installation and
+# package installation) and the desire to install cleanly via pip (instead of directly with
+# setuptools), this seems to be the best configuration possible
+skip_install = true
+commands =
+    pip install --no-deps --editable . --editable clients/python
+    sphinx-build \
+        -W \
+        -b html \
+        -d {envtmpdir}/doctrees \
+        docs \
+        build/docs/python
```

Comment:
Does it make sense to pass `{posargs`} here?

Reply:
Good question. What other options would someone want to inject though? Running a different builder would be a common use case, but you'd have to retype the command manually, since a builder is already hardcoded. The only few that I see are:
`-E                don't use a saved environment, always read all files`
`-j N              build in parallel with N processes where possible`
`-n                nit-picky mode, warn about all missing references`
`-v                increase verbosity (can be repeated)`
Some of these actually may be interesting enough to turn on all the time, but I feel like the others are getting so low-level that if a developer wants to specify them, it's probably time to run the command manually.

Reply:
Looking at that list, I would vote for `-n` to be added to the defaults. It seems difficult to justify it since I don't build the docs quite often, but any/all of those options and more (full tracebacks, pdb, only showing warnings, configuration settings, etc).

Reply:
Unrelated, but for anyone passing by - I recently discovered https://github.com/GaretJax/sphinx-autobuild which makes the process of editing docs with sphinx significantly less painful.

Reply:
@danlamanna We'll have to address this in the future. See #2643 and #2646 .